bitbots_head_mover/src/move_head.cpp

```diff
@@ -0,0 +1,679 @@
+                                     params_.search_pattern_penalty.tilt_max[1],
+                                     params_.search_pattern.reduce_last_scanline);
+          break;
+
+        case humanoid_league_msgs::msg::HeadMode::FIELD_FEATURES: // 3
+          pan_speed_ = params_.search_pattern_field_features.pan_speed;
+          tilt_speed_ = params_.search_pattern_field_features.tilt_speed;
+          pattern_ = generatePattern(params_.search_pattern_field_features.scan_lines,
+                                     params_.search_pattern_field_features.pan_max[0],
+                                     params_.search_pattern_field_features.pan_max[1],
+                                     params_.search_pattern_field_features.tilt_max[0],
+                                     params_.search_pattern_field_features.tilt_max[1],
+                                     params_.search_pattern.reduce_last_scanline);
+          break;
+
+        case humanoid_league_msgs::msg::HeadMode::LOOK_FRONT: // 13
+          pan_speed_ = params_.front_search_pattern.pan_speed;
+          tilt_speed_ = params_.front_search_pattern.tilt_speed;
+          pattern_ = generatePattern(params_.front_search_pattern.scan_lines,
+                                     params_.front_search_pattern.pan_max[0],
+                                     params_.front_search_pattern.pan_max[1],
+                                     params_.front_search_pattern.tilt_max[0],
+                                     params_.front_search_pattern.tilt_max[1],
+                                     params_.search_pattern.reduce_last_scanline);
+          break;
+
+          case humanoid_league_msgs::msg::HeadMode::LOOK_FORWARD: // 7
+          pan_speed_ = params_.look_forward.pan_speed;
+          tilt_speed_ = params_.look_forward.tilt_speed;
+          pattern_ = generatePattern(params_.look_forward.scan_lines,
+                                     params_.look_forward.pan_max[0],
+                                     params_.look_forward.pan_max[1],
+                                     params_.look_forward.tilt_max[0],
+                                     params_.look_forward.tilt_max[1],
+                                     params_.search_pattern.reduce_last_scanline);
+          break;
+
+        default:
+          return;
+      }
+      std::pair<double, double> head_position = get_head_position();
+
+      index_ = get_near_pattern_position(pattern_, head_position.first, head_position.second);
+    }
+    if (!action_running_ && curr_head_mode != humanoid_league_msgs::msg::HeadMode::DONT_MOVE){ //here DONT_MOVE is implemented
+      prev_head_mode_ = curr_head_mode; 
+      perform_search_pattern();
+
+    }
+  };
+
+  std::shared_ptr<rclcpp::Node> get_node() {
+    return node_;
+  }
+};}
+
+int main(int argc, char *argv[]) {
+  rclcpp::init(argc, argv);
+  auto head_mover = std::make_shared<move_head::HeadMover>();
+  rclcpp::spin(head_mover->get_node());
```

Comment:
Maybe use the events executor instead of the normal one. See e.g. convenience frames.